miner/Miner.hs

```diff
@@ -208,107 +191,114 @@ main = do
     opts = info (pClientArgs <**> helper)
         (fullDesc <> progDesc "The Official Chainweb Mining Client")
 
-scheme :: Env -> (TargetBytes -> HeaderBytes -> IO HeaderBytes)
-scheme env = case cmd $ args env of
-    CPU e -> cpu (version $ args env) e (gen env)
-    GPU _ -> gpu
+run :: RIO Env ()
+run = do
+    logInfo "Starting Miner."
+    env <- ask
+    case cmd $ args env of
+        GPU -> logError "GPU mining is not yet available."
+        CPU _ -> getWork >>= traverse_ (mining (scheme env))
+    liftIO exitFailure
```

Comment:
The miner is designed to never exit. If mining ever returns, we should consider it a failure, hence `exitFailure`.